sklearn/base.py

```diff
@@ -125,7 +138,17 @@ def _pprint(params, offset=0, printer=repr):
     return lines
 
 
-###############################################################################
+def _update_if_consistent(dict1, dict2):
+    common_keys = set(dict1.keys()).intersection(dict2.keys())
+    for key in common_keys:
+        if dict1[key] != dict2[key]:
+            raise TypeError("Inconsistent values for tag {}: {} != {}".format(
```

Comment:
Right, but then this would error if the `BaseEstimator` and the final estimator class define the same tag? Maybe re-ordering slightly the MRO classes (cf https://github.com/scikit-learn/scikit-learn/pull/8022#discussion_r234749506), then not overwriting existing tags could be a way around it...

Reply:
yes, which I solved by having the ``BaseEstimator`` not defining anything. The current solutions does what I want...

Reply:
(I think)